### PR TITLE
feat: Add blueprint connections and related metadata

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

File: cli/bpmetadata/proto/bpmetadata.proto [hunk 1]
```diff
@@ -453,6 +468,43 @@ message BlueprintVariable {
   string var_type = 3; // @gotags: json:"varType,omitempty" yaml:"varType,omitempty"
   google.protobuf.Value default_value = 4; // @gotags: json:"defaultValue,omitempty" yaml:"defaultValue,omitempty"
   bool required = 5; // @gotags: json:"required,omitempty" yaml:"required,omitempty"
+  // Incoming connections to this variable.
```
Comment (g-awmalik) [inline cli/bpmetadata/proto/bpmetadata.proto:471]: Can we define "connections" a bit more explicitly so it's easier for a someone without ADC context to understand?

File: cli/bpmetadata/proto/bpmetadata_ui.proto [hunk 2]
```diff
@@ -126,6 +127,22 @@ message DisplayVariable {
   // Optional.
   // Gen: manually-authored
   string boolean_group = 20; // @gotags: json:"booleanGroup,omitempty" yaml:"booleanGroup,omitempty"
+
+  // Alternate default value.
```
Comment (g-awmalik) [inline cli/bpmetadata/proto/bpmetadata_ui.proto:131]: What is this used for?

**Review by g-awmalik — COMMENTED** [review]
LGTM for the most part but a few concepts require a bit more description.

**Comment by g-awmalik** [comment]
@njlisi - please review.

File: cli/bpmetadata/proto/bpmetadata.proto (hunk 1 above)
Comment (bharathkkb) [reply cli/bpmetadata/proto/bpmetadata.proto:471]: good point, added some more details

File: cli/bpmetadata/proto/bpmetadata_ui.proto (hunk 2 above)
Comment (bharathkkb) [reply cli/bpmetadata/proto/bpmetadata_ui.proto:131]: Added a line with expected usage.

File: cli/bpmetadata/proto/bpmetadata_ui.proto [hunk 3]
```diff
@@ -126,6 +127,23 @@ message DisplayVariable {
...
+  // This allows authors to define an alternative value for pre identified usecases such as security.
```
Comment (njlisi) [inline cli/bpmetadata/proto/bpmetadata_ui.proto:132]: Could you add some more context here? "AlternateDefault" name gives an impression that this field will provide a different value in place of the _default_value_ field in _BlueprintVariable_(bpmetadata.proto), is that correct? Is it used for providing supporting information for the default_value?

**Review by njlisi — COMMENTED** [review]
LGTM

File: cli/bpmetadata/proto/bpmetadata_ui.proto [hunk 4]
```diff
@@ -126,6 +127,23 @@ message DisplayVariable {
...
+  // Gen: manually-authored.
+  message AlternateDefault {
+    // Type of the alternate default.
+    enum AlternateType {
```
Comment (q2w) [inline cli/bpmetadata/proto/bpmetadata_ui.proto:136]: Going forward if a new alternate type is needed, will we update this enum? Let's say if we want a default for ADC use case, would we created new alternate type?

**Review by q2w — APPROVED** [review]

File: cli/bpmetadata/proto/bpmetadata_ui.proto (hunk 3 above)
Comment (bharathkkb) [reply cli/bpmetadata/proto/bpmetadata_ui.proto:132]: Added some more info

> "AlternateDefault" name gives an impression that this field will provide a different value in place of the default_value field in BlueprintVariable(bpmetadata.proto), is that correct?

Yup! Its intended to be a hand authored alternate to the one in BlueprintVariable.

File: cli/bpmetadata/proto/bpmetadata_ui.proto (hunk 4 above)
Comment (bharathkkb) [reply cli/bpmetadata/proto/bpmetadata_ui.proto:137]: Yes and I am thinking ADC can also able to leverage or surface some these existing alternates

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**